src/libcore/str/mod.rs

```diff
@@ -34,99 +36,33 @@ use result::Result::{self, Ok, Err};
 use slice::{self, SliceExt};
 use usize;
 
-pub use self::pattern::Pattern;
-pub use self::pattern::{Searcher, ReverseSearcher, DoubleEndedSearcher, SearchStep};
+pub mod pattern;
 
-mod pattern;
+macro_rules! iter_methods {
+    (: $te:ty) => {
```

Comment:
In general I've been somewhat uncomfortable with how many macros are being used across all these impls, it ends up making it more difficult to read I think as you have to keep referring back to what the macros are actually doing. This is actually a pretty small macro now, so could it just be inlined manually now that all the `impl` blocks are explicit?